src/foremast/datapipeline/datapipeline.py

```diff
@@ -0,0 +1,109 @@
+        self.env = env
+        self.region = region
+        self.properties = get_properties(prop_path)
+        self.datapipeline_data = self.properties[self.env]['datapipeline']
+        generated = get_details(app=self.app_name)
+        self.group = generated.data['project']
+
+        session = boto3.Session(profile_name=self.env, region_name=self.region)
+        self.client = session.client('datapipeline')
+        self.pipeline_id = None
+
+    def create_datapipeline(self):
+        """Creates the data pipeline if it does not already exist"""
+
+        tags = [
+                {"key": "app_group",
+                "value": self.group},
+                {"key": "app_name",
+                "value": self.app_name}
+                ]
+        response = self.client.create_pipeline(name=self.datapipeline_data.get('name', self.app_name),
+                                               uniqueId=self.app_name,
+                                               description=self.datapipeline_data['description'],
+                                               tags=tags)
+        self.pipeline_id = response.get('pipelineId')
+        LOG.debug(response)
+        LOG.info("Successfully configured Data Pipeline - %s", self.app_name)
+
+    def set_pipeline_definition(self):
+        """Translates the json definition and puts it on created pipeline"""
+
+        if not self.pipeline_id:
+            self.get_pipeline_id()
+
+        pipelineObjects = translator.definition_to_api_objects(self.datapipeline_data['json_definition'])
+        parameterObjects = translator.definition_to_api_parameters(self.datapipeline_data['json_definition'])
+        parameterValues = translator.definition_to_parameter_values(self.datapipeline_data['json_definition'])
+        response = self.client.put_pipeline_definition(
+                                pipelineId=self.pipeline_id,
+                                pipelineObjects=pipelineObjects,
+                                parameterObjects=parameterObjects,
+                                parameterValues=parameterValues)
+        LOG.debug(response)
+        LOG.info("Successfully applied pipeline definition")
+
+    def get_pipeline_id(self):
+        """Finds the pipeline ID for configured pipeline"""
+
+        all_pipelines = []
+        hasMore = True
+        marker = None
+        # handles the pagination from boto3
+        while hasMore:
+            if not marker:
+                response = self.client.list_pipelines()
+            else:
+                LOG.info("Checking for more pipelines")
+                response = self.client.list_pipelines(marker=marker)
+            all_pipelines.extend(response['pipelineIdList'])
+            hasMore = response['hasMoreResults']
+            marker = response.get('marker')
+            LOG.debug(response)
+        
+        for pipeline in all_pipelines:
+            if pipeline['name'] == self.datapipeline_data.get('name', self.app_name):
```

Comment:
would this really be any better? Currently I exit out early if found and I never need to relook it up. I think this would add complexity